### PR TITLE
Improve bug report issue template language

### PR DESCRIPTION
**Pull request by alexcreasy**
## Description

Makes language on our bug template a bit clearer when filing bugs against unreleased feature branches.

## How Has This Been Tested?

N/A text only

## Test Impact

None

## Request review criteria:

Read the text changes and give an opinion.

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`
^^^ **Textual change to GitHub templates, above is N/A**

### CODE REVIEW

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ body:
     id: version
     attributes:
       label: Version
-      description: What was the version this was found on?
+      description: What was the version this was found on (use branch name if unreleased)?
```
Comment (andrewballantyne) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:33]: Maybe we alter this to give more examples?

```suggestion
      description: |-
        What was the version this was found on?
        
        eg. a branch name, a commit id, or a version number
```

**Comment by alexcreasy** [comment]
Let me just squash this and we should be good to go

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Review by manaswinidas — APPROVED** [review]
/lgtm

**Pull request merged**